### PR TITLE
New Parser for rdma config file

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: insights/parsers/rdma_config.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        # Should we modify the system mtrr registers?  We may need to do this if you
+        # get messages from the ib_ipath driver saying that it couldn't enable
+        # write combining for the PIO buffs on the card.
+        #
+        # Note: recent kernels should do this for us, but in case they don't, we'll
+        # leave this option
+        FIXUP_MTRR_REGS=no
+
+    Examples:
+        >>> rdma_conf = shared[RdmaConfig]
```
Comment (xiangce) [inline insights/parsers/rdma_config.py]: `shared` is deprecated, it not recommended to it in the `Examples` which is also a guide to uses. Please update it as per the [sysconfig](https://github.com/RedHatInsights/insights-core/blob/master/insights/parsers/sysconfig.py) and add the corresponding doctest by referring to https://github.com/RedHatInsights/insights-core/blob/master/insights/parsers/tests/test_sysconfig_doc_examples.py#L152

File: insights/parsers/rdma_config.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        'yes'
+        >>> rdma_conf["SRP_LOAD"]
+        'yes'
+        >>> rdma_conf["SVCRDMA_LOAD"]
+        'no'
+
+    """
+
+    def parse_content(self, content):
+        self.data = split_kv_pairs(get_active_lines(content))
```
Comment (xiangce) [inline insights/parsers/rdma_config.py]: Better ignore the empty content, e.g. raise a `SkipException` when the `self.data` is empty.

And test it, see https://github.com/RedHatInsights/insights-core/blob/master/insights/parsers/tests/test_ssh_client_config.py#L63

**Review by xiangce — CHANGES_REQUESTED** [review]

File: insights/parsers/rdma_config.py (hunk 2 above)
Comment (rasrivas-redhat) [reply insights/parsers/rdma_config.py]: @xiangce - updated the changes

File: insights/parsers/rdma_config.py (hunk 1 above)
Comment (rasrivas-redhat) [reply insights/parsers/rdma_config.py]: @xiangce - updated the changes

File: insights/parsers/rdma_config.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        >>> rdma_conf["SRP_LOAD"]
+        'yes'
+        >>> rdma_conf["SVCRDMA_LOAD"]
+        'no'
+    """
+
+    def parse_content(self, content):
+        self.data = split_kv_pairs(get_active_lines(content))
+
+        _content = get_active_lines(content)
```
Comment (xiangce) [inline insights/parsers/rdma_config.py]: Better move this code to the very beginning of the `parse_content`.
And replace the calling of `get_active_lines(content)` to `_content`.

Comment (rasrivas-redhat) [reply insights/parsers/rdma_config.py]: updated the changes as per your suggestion

**Review by xiangce — APPROVED** [review]

**Comment by rasrivas-redhat** [comment]
@xiangce - can you please merge this branch

File: insights/parsers/rdma_config.py [hunk 4]
```diff
@@ -2,13 +2,14 @@
...
-from .. import get_active_lines, Parser, parser, LegacyItemAccess
+from .. import get_active_lines, Parser, parser
 from insights.parsers import split_kv_pairs
 from insights.specs import Specs
+from insights.parsers import SkipException
 
 
 @parser(Specs.rdma_conf)
-class RdmaConfig(Parser, LegacyItemAccess):
+class RdmaConfig(Parser):
```
Comment (xiangce) [inline insights/parsers/rdma_config.py]: Just noticed that you removed the `LegacyItemAccess`. It's not good to do so.

Comment (rasrivas-redhat) [reply insights/parsers/rdma_config.py]: updated the changes

**Review by xiangce — APPROVED** [review]

**Pull request merged**